### PR TITLE
Fix NPE in collecting success counts

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: src/borda_clj/client.clj [hunk 1]
```diff
@@ -15,10 +15,11 @@
    caps the size of the hashmap at the given max-buffer-size."
   [measurements max-buffer-size dimensions values]
   (let [vals (merge {:_submits 1} values)]
+    (println "collecting into" measurements)
     (if (contains? measurements dimensions)
       ; TODO: support stuff other than SUM here (e.g. AVG, MIN, MAX, etc.)
       (update measurements dimensions (partial merge-with +) vals) ; merge with existing in buffer
-      (if (< (count measurements) max-buffer-size)
+      (if (or (= submit-key dimensions) (< (count (dissoc measurements submit-key)) max-buffer-size))
```
Comment (oxtoacart) [inline src/borda_clj/client.clj]: This exempts the submit-key from our buffer size.

File: test/borda_clj/client_test.clj [hunk 2]
```diff
@@ -22,20 +22,22 @@
...
                           (do (reset! fail false) (throw (Exception. "I'm failing"))) ; fail on first send
                           (swap! result (fn [orig] (if (= orig {}) next orig)))))     ; update on subsequent send
-        log-error     (fn [measurements e] (println "error on sending" (count measurements) "measurements to borda"))
-        [submit stop] (reducing-submitter global 10 100 update log-error)]
+        log-error     (fn [measurements e] (println "error on sending" (count measurements) "measurements to borda" e))
+        [submit stop] (reducing-submitter global 1 100 update log-error)]
     (testing "Submit works"
       (submit da va)
       (submit da vb)
+      (submit db va) ; this should be discarded because buffer is full
```
Comment (oxtoacart) [inline test/borda_clj/client_test.clj]: Added this to test the buffer full condition.

**Comment by aranhoide** [comment]
LGTM!

**Comment by oxtoacart** [comment]
Thanks @aranhoide!

**Pull request merged**